system_tests/espresso_e2e_test.go

```diff
@@ -165,4 +277,74 @@ func TestEspressoE2E(t *testing.T) {
 		return validatedCnt >= msgCnt
 	})
 	Require(t, err)
+
+	hostIo, err := ospgen.NewOneStepProverHostIo(common.HexToAddress(hostIoAddress), builder.L1.Client)
+	Require(t, err)
+	actualCommitment, err := hostIo.GetHotShotCommitment(&bind.CallOpts{}, big.NewInt(1))
+	Require(t, err)
+	commitmentBytes := actualCommitment.Bytes()
+	if len(commitmentBytes) != 32 {
+		t.Fatal("failed to read hotshot via hostio contract, length is not 32")
+	}
+	empty := true
+	for _, i := range commitmentBytes {
+		if i != 0 {
+			empty = false
+			break
+		}
+	}
+	if empty {
+		t.Fatal("failed to read hotshot via hostio contract, empty")
+	}
+	log.Info("Read hotshot commitment via hostio contract successfully", "height", 1, "commitment", commitmentBytes)
+
+	lastValidatedInfo, err := node.ConsensusNode.BlockValidator.ReadLastValidatedInfo()
+	Require(t, err)
+	incorrectHeight := lastValidatedInfo.GlobalState.HotShotHeight
+	log.Info("setting incorrect hotshot height!", "height", incorrectHeight)
+	validated := node.ConsensusNode.BlockValidator.Validated(t)
+
+	goodStaker, blockValidatorA, cleanA := createStaker(ctx, t, builder, 0)
```

Comment:
I think that this interface would be cleaner if you passed in something like 

`EspressoDebugParams {
  divergeHeight: 0 
}`